userspace/libsinsp/parsers.cpp

```diff
@@ -1189,22 +1189,27 @@ void sinsp_parser::parse_clone_exit(sinsp_evt *evt)
 		tinfo.m_pid = childtid;
 	}
 
-	//
-	// Copy the fd list
-	// XXX this is a gross oversimplification that will need to be fixed.
-	// What we do is: if the child is NOT a thread, we copy all the parent fds.
-	// The right thing to do is looking at PPM_CL_CLONE_FILES, but there are
-	// syscalls like open and pipe2 that can override PPM_CL_CLONE_FILES with the O_CLOEXEC flag
-	//
 	if(!(tinfo.m_flags & PPM_CL_CLONE_THREAD))
 	{
+		//
+		// Copy the fd list
+		// XXX this is a gross oversimplification that will need to be fixed.
+		// What we do is: if the child is NOT a thread, we copy all the parent fds.
+		// The right thing to do is looking at PPM_CL_CLONE_FILES, but there are
+		// syscalls like open and pipe2 that can override PPM_CL_CLONE_FILES with the O_CLOEXEC flag
+		//
 		tinfo.m_fdtable = *(ptinfo->get_fd_table());
 
 		//
 		// It's important to reset the cache of the child thread, to prevent it from
 		// referring to an element in the parent's table.
 		//
 		tinfo.m_fdtable.reset_cache();
+
+		//
+		// Not a thread, copy cwd
+		//
+		tinfo.m_cwd = ptinfo->m_cwd;
```

Comment:
FYI, this could be in a separate `if` that checks for `PPM_CL_CLONE_FS`. I (over)simplified this assumption and collapsed it with `CLONE_THREAD` based on 3 things:

1) We already do the exact same thing for the whole fd table, which is far more important than the cwd, and nobody complained yet

2) `libc` will always create the threads passing both `CLONE_THREAD` and `CLONE_FS`. From the glibc code of `start_thread` (called with `pthread_create`):

```
  const int clone_flags = (CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SYSVSEM
			   | CLONE_SIGHAND | CLONE_THREAD
			   | CLONE_SETTLS | CLONE_PARENT_SETTID
			   | CLONE_CHILD_CLEARTID
			   | 0);

```

3) I was dubious about using CLONE_FS flag because we don't propagate it in libscap even if we probably should, and I didn't want to create further damage with this patch: https://github.com/draios/sysdig/blob/dev/userspace/libscap/scap_procs.c#L517

Very interested to hear other opinions